### PR TITLE
torchmetrics: bump min version v0.10 in examples

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Note this is rather for our maintenance as we do now need to deal with the `task` arguments in classification metrics as it was mostly already updated in #15904
But also fine to bump the base requirements too :)

<details>
  <summary><b>Before submitting</b></summary>

- Was this **discussed/agreed** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- Did you make sure to **update the documentation** with your changes? (if necessary)
- Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- Did you list all the **breaking changes** introduced by this pull request?
- Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

</details>

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

<details>
  <summary>Reviewer checklist</summary>

- [ ] Is this pull request ready for review? (if not, please submit in draft mode)
- [ ] Check that all items from **Before submitting** are resolved
- [ ] Make sure the title is self-explanatory and the description concisely explains the PR
- [ ] Add labels and milestones (and optionally projects) to the PR so it can be classified

</details>

cc @borda @tchaton @carmocca @justusschock @awaelchli

### CODE REVIEW

**Comment by Borda** [comment]
PL import is handled in #16935

File: examples/pl_domain_templates/computer_vision_fine_tuning.py [hunk 1]
```diff
@@ -186,8 +186,8 @@ def __init__(
 
         self.__build_model()
 
-        self.train_acc = Accuracy()
-        self.valid_acc = Accuracy()
+        self.train_acc = Accuracy(task="multiclass")
+        self.valid_acc = Accuracy(task="multiclass")
```
Comment (awaelchli) [inline examples/pl_domain_templates/computer_vision_fine_tuning.py:190]: Here it's binary classification

Comment (Borda) [reply examples/pl_domain_templates/computer_vision_fine_tuning.py:190]:
```suggestion
        self.train_acc = Accuracy(task="binary")
        self.valid_acc = Accuracy(task="binary")
```

Comment (Borda) [reply examples/pl_domain_templates/computer_vision_fine_tuning.py:190]:
```suggestion
        self.train_acc = Accuracy(task="binary")
        self.valid_acc = Accuracy(task="binary")
```

**Review by awaelchli — APPROVED** [review]

**Review by justusschock — APPROVED** [review]

**Pull request merged**